app/src/main/java/fr/free/nrw/commons/contributions/ContributionsListFragment.java

```diff
@@ -120,8 +120,12 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
         if (resultCode == RESULT_OK) {
             Timber.d("OnActivityResult() parameters: Req code: %d Result code: %d Data: %s",
                     requestCode, resultCode, data);
-            Uri uri = ContributionUtils.saveFileBeingUploadedTemporarily(getActivity(), data.getData());
-            controller.handleImagePicked(requestCode, uri, false, null);
+            if (requestCode == ContributionController.SELECT_FROM_CAMERA) {
+                controller.handleImagePicked(requestCode, null, false, null);
```

Comment:
@neslihanturan Why is the `uri` being passed as null for camera clicks?

Reply:
Also, please update the `handleImagePicked` method signature annotating this param as `@Nullable` :)

Reply:
It is null because, data coming from camera intent is already null. Instead, photo gets saved to a directory and its path will be get from intent extras on subsequent steps.